src/or/nodelist.c

```diff
@@ -1134,7 +1134,7 @@ node_is_dir(const node_t *node)
 /** Return true iff <b>node</b> has either kind of usable descriptor -- that
  * is, a routerdescriptor or a microdescriptor. */
```

Comment:
Comment: You should probably use node_has_preferred_descriptor() instead of this function.

Reply:
f5a472c20da9a0 does that.